transform-plugins/src/test/java/co/cask/hydrator/plugin/ValueMapperTest.java

```diff
@@ -604,107 +329,81 @@ public void testWithMultipleMapping() throws Exception {
-      .build();
+  }
 
+  @Test
+  public void testSchemaHandling() throws Exception {
 
-    AppRequest<ETLBatchConfig> appRequest = new AppRequest<>(ETLBATCH_ARTIFACT, etlConfig);
-    Id.Application appId = Id.Application.from(Id.Namespace.DEFAULT, "valuemappertest_seven");
-    ApplicationManager appManager = deployApplication(appId, appRequest);
+    Schema inputSchema = Schema.recordOf("sourceRecord",
+                                         Schema.Field.of(ID, Schema.of(Schema.Type.STRING)),
+                                         Schema.Field.of(NAME, Schema.of(Schema.Type.STRING)),
+                                         Schema.Field.of(SALARY, Schema.of(Schema.Type.STRING)),
+                                         Schema.Field.of(DESIGNATIONID, Schema.unionOf(Schema.of(Schema.Type.STRING),
+                                                                                       Schema.of(Schema.Type.NULL))));
 
-    addDatasetInstance(KeyValueTable.class.getName(), "designation_lookup_table_seven");
-    DataSetManager<KeyValueTable> dataSetManager = getDataset("designation_lookup_table_seven");
-    KeyValueTable keyValueTable = dataSetManager.get();
-    keyValueTable.write("1".getBytes(Charsets.UTF_8), "SE".getBytes(Charsets.UTF_8));
-    keyValueTable.write("2".getBytes(Charsets.UTF_8), "SSE".getBytes(Charsets.UTF_8));
-    keyValueTable.write("3".getBytes(Charsets.UTF_8), "ML".getBytes(Charsets.UTF_8));
-    dataSetManager.flush();
+    ValueMapper.Config config = new ValueMapper.Config("designationid:designation_lookup_table:designationName",
+                                                       "designationid:DEFAULTID");
 
-    DataSetManager<Table> inputManager = getDataset(inputTable);
-    List<StructuredRecord> input = ImmutableList.of(
-      StructuredRecord.builder(SOURCE_SCHEMA).set(ID, "100").set(NAME, "John").set(SALARY, "1000").set
-        (DESIGNATIONID, "1").build(),
-      StructuredRecord.builder(SOURCE_SCHEMA).set(ID, "101").set(NAME, "Kerry").set(SALARY, "1030").set
-        (DESIGNATIONID, "2").build(),
-      StructuredRecord.builder(SOURCE_SCHEMA).set(ID, "102").set(NAME, "Mathew").set(SALARY, "1230").set
-        (DESIGNATIONID, "3").build()
-    );
-    MockSource.writeInput(inputManager, input);
+    MockPipelineConfigurer configurer = new MockPipelineConfigurer(inputSchema);
+    new ValueMapper(config).configurePipeline(configurer);
+    Schema outputSchema = configurer.getOutputSchema();
 
-    MapReduceManager mrManager = appManager.getMapReduceManager(ETLMapReduce.NAME);
-    mrManager.start();
-    mrManager.waitForFinish(5, TimeUnit.MINUTES);
 
-    DataSetManager<Table> outputManager = getDataset(sinkTable);
-    List<StructuredRecord> outputRecords = MockSink.readOutput(outputManager);
-    Map<String, String> nameDesignationMap = new HashMap<String, String>();
-    nameDesignationMap.put("John", "SE");
+    Assert.assertEquals(Schema.of(Schema.Type.STRING).getType(),
```

Comment:
you should assert the entire Schema objects are equal rather than just checking a single field type.  You can just use the schema object directly, don't need to check field by field.